compiler/lib/src/main/scala/analysis/Semantics/Event.scala

```diff
@@ -35,14 +45,59 @@ object Event {
         )
         else Right(())
       }
+      def getEveryIntervalValue(every: AstNode[Ast.Expr]) = {
+        val loc = Locations.get(every.id)
+        val Value.AnonStruct(intervalValue) = Analysis.convertValueToType(a.valueMap(every.id), Type.AnonStruct(
+          Map(
+            ("seconds", Type.U32),
+            ("useconds", Type.U32),
+          )
+        ))
+        def getNonZeroMember(member: String, maxValue: Option[Int] = None) = {
```

Comment:
I think this function should not be called `getNonZeroMember`. Zero is allowed. Maybe just `getMember`?